### ISSUE TITLE
New Rule: no-unhandled-nullable-return

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#47]
## Research Evidence

From empirical LLM bug studies: LLMs often miss corner cases, including null/undefined returns from functions.

## Description

Detect when a function that can return null/undefined is called but the result is used without null checking.

### Bad
```ts
function findUser(id: string): User | undefined { ... }

const user = findUser('123');
user.name; // TypeScript catches this, but LLMs might bypass with !
```

### Also Bad (LLM uses non-null assertion)
```ts
const user = findUser('123')!;
console.log(user.name); // LLM added ! to silence type error
```

### Good
```ts
const user = findUser('123');
if (!user) {
  throw new Error('User not found');
}
console.log(user.name);
```

## Focus
This rule would specifically catch non-null assertions (`!`) on nullable returns, which is how LLMs "solve" type errors without proper handling.

## Priority
Medium - TypeScript already catches, but ! bypass is common LLM pattern
